types/model/name.go

```diff
@@ -686,7 +686,10 @@ func IsValidNamePart(kind PartKind, s string) bool {
 		return false
 	}
 	var consecutiveDots int
-	for _, c := range []byte(s) {
+	for i, c := range []byte(s) {
+		if i == 0 && !isAlphaNumeric(c) {
+			return false
+		}
 		if c == '.' {
```

Comment:
Is stopping `..` important? GitHub lets you do it. I'm okay either way

Reply:
Yes. We convert these to filepaths and need to take measure to prevent path traversals or unexpected short filepaths.

Reply:
sounds good, that seems more prudent!